pkg/k8s/shared_resources.go

```diff
@@ -42,15 +42,15 @@ var (
 
 type SharedResources struct {
 	cell.In
-	LocalNode       *LocalNodeResource
-	LocalCiliumNode *LocalCiliumNodeResource
-	Services        resource.Resource[*slim_corev1.Service]
-	Namespaces      resource.Resource[*slim_corev1.Namespace]
-	LBIPPools       resource.Resource[*cilium_api_v2alpha1.CiliumLoadBalancerIPPool]
-	Identities      resource.Resource[*cilium_api_v2.CiliumIdentity]
+	LocalNode       cell.Optional[LocalNodeResource]
+	LocalCiliumNode cell.Optional[LocalCiliumNodeResource]
+	Services        cell.Optional[resource.Resource[*slim_corev1.Service]]
+	Namespaces      cell.Optional[resource.Resource[*slim_corev1.Namespace]]
+	LBIPPools       cell.Optional[resource.Resource[*cilium_api_v2alpha1.CiliumLoadBalancerIPPool]]
+	Identities      cell.Optional[resource.Resource[*cilium_api_v2.CiliumIdentity]]
```

Comment:
While I like the semantic hint that `cell.Optional` gives, I'm still grimacing hard when I look at these types. Perhaps for things like `Resource` we could solve this in another way (e.g. Events() closing immediately and Store() rejecting the promise), or at the very least type alias these somehow. WDYT?

Reply:
Making an alias of a generic isn't allowed by Go at the moment, there is a proposal which might land in 1.21 https://github.com/golang/go/issues/46477

So we have to accept the sub-optimal syntax or back-track on doing it this way and going the "cells are never nil" route and providing types that always error if certain features are disabled.